backend/workflow_manager/workflow/execution.py

```diff
@@ -198,36 +198,41 @@ def execute(self, single_step: bool = False) -> None:
-            if (
-                self.execution_mode == WorkflowExecution.Mode.INSTANT
-                or self.execution_mode == WorkflowExecution.Mode.QUEUE
-            ):
-                start_time = time.time()
-                try:
-                    self.execute_workflow(execution_type=execution_type)
-                    end_time = time.time()
-                    execution_time = end_time - start_time
-                except StopExecution as exception:
-                    end_time = time.time()
-                    execution_time = end_time - start_time
-                    logger.info(f"Execution {self.execution_id} stopped")
-                    raise exception
-                except Exception as exception:
-                    end_time = time.time()
-                    execution_time = end_time - start_time
-                    message = str(exception)[:EXECUTION_ERROR_LENGTH]
-                    logger.info(
-                        f"Execution {self.execution_id} in {execution_time}s, "
-                        f" Error {exception}"
-                    )
-                    raise WorkflowExecutionError(message) from exception
-            else:
-                error_message = f"Unknown Execution Method {self.execution_mode}"
-                raise WorkflowExecutionError(error_message)
+        if self.compilation_result["success"] is False:
+            error_message = (
+                f"Errors while compiling workflow "
+                f"{self.compilation_result['problems'][0]}"
+            )
+            raise WorkflowExecutionError(error_message)
 
+        if self.execution_mode not in (
+            WorkflowExecution.Mode.INSTANT,
+            WorkflowExecution.Mode.QUEUE,
+        ):
+            error_message = f"Unknown Execution Method {self.execution_mode}"
+            raise WorkflowExecutionError(error_message)
+
+        start_time = time.time()
+        try:
+            self.execute_workflow(execution_type=execution_type)
+            end_time = time.time()
+            execution_time = end_time - start_time
+        except StopExecution as exception:
+            end_time = time.time()
+            execution_time = end_time - start_time
+            logger.info(f"Execution {self.execution_id} stopped")
+            raise exception
+        except Exception as exception:
+            end_time = time.time()
+            execution_time = end_time - start_time
+            message = str(exception)[:EXECUTION_ERROR_LENGTH]
+            logger.info(
+                f"Execution {self.execution_id} in {execution_time}s, "
+                f" Error {exception}"
+            )
+            raise WorkflowExecutionError(message) from exception
         else:
-            error_message = f"Errors while compiling workflow {self.compilation_result['problems'][0]}"  # noqa
+            error_message = f"Unknown Execution Method {self.execution_mode}"
             raise WorkflowExecutionError(error_message)
```

Comment:
@chandrasekharan-zipstack This needs to be removed. It is checked after final change, right?

Reply:
@Deepak-Kesavan thanks for catching this, I had committed from the UI and missed this. Updated it